### DISCUSSION TITLE
Persist with multiple slices and typescript

### DISCUSSION THREAD

**Discussion by dvillegasGlobalis** [pmndrs/zustand#2451] [Q&A]
I can't find anywhere this combination, and also with partialize. In the docs you have the example of multiple slices with persist but without typescript, and multiple slices with typescript but without persist. Currently i have this code.

```
export const useStore = create<UserSlice>()((...a) => ({
  ...persist(createUserSlice, {
    name: 'user-storage',
    storage: createJSONStorage(() => localStorage),
    partialize: (state) => ({
      token: state.token,
      refToken: state.refToken,
      currentUser: state.currentUser
    })
  })(...a)
}));

```

This works great but i don't know how to add another slice with this structure. I need to add a field of the other slice to the partialize.

This is the other slice that i want to add. And i need to add the property currentWorkspace to partialize.

```
export interface AppSlice {
  currentWorkspace?: string;
  setCurrentWorkspace: (workspace: string) => void;
}

export const createAppSlice: StateCreator<AppSlice> = (set) => ({
  currentWorkspace: undefined,
  setCurrentWorkspace: (workspace: string) => {
    set({ currentWorkspace: workspace });
  },
});
```

**Comment by Simoon-F** [comment]
It should be like this：

```ts
export const useStore = create<UserSlice>()((...a) => ({
  ...persist(createUserSlice, {
    name: 'user-storage',
    storage: createJSONStorage(() => localStorage),
    partialize: (state) => ({
      token: state.token,
      refToken: state.refToken,
      currentUser: state.currentUser
    })
  })(...a),

  // add
  ...createXxxSlice(...a),
}));
```

Update:

I saw an issues, which may have a better answer. The above example may not be appropriate.

https://github.com/pmndrs/zustand/discussions/2437#discussioncomment-8921356

**Comment by dbritto-dev** [comment]
@dvillegasGlobalis @Simoon-F just to let you know middlewares shouldn't be applied to slices but only whole stores

**Reply by Simoon-F** [reply]
> @dvillegasGlobalis @Simoon-F just to let you know middlewares shouldn't be applied to slices but only whole stores只是为了让你知道中间件不应该应用于切片，而应该只应用于整个商店

@dbritto-dev  but some slices do not need persist middleware.

**Reply by dbritto-dev** [reply]
@Simoon-F the best ways to persist just some fields is by using `partialize` or by creating a custom storage. But keep in mind that middlewares work at store level and you shouldn't use it on slices directly

**Reply by Simoon-F** [reply]
@dbritto-dev Okay, got it~
